lib/index.js

```diff
@@ -68,7 +68,7 @@ export const _dotProp = (string, object) => string
   .split('.')
   .reduce((acc, key) => acc[key], object);
 
-export const mapToTheme = (key) => 
-  (props) => styledMap(_dotProp(key, props.theme));
+export const mapToTheme = (key, prop) =>
+  (props) => prop ? styledMap(prop, _dotProp(key, props.theme)) : styledMap(_dotProp(key, props.theme));
```

Comment:
Could you add the prop parameter to the typings file? I think `(key: string, prop?: string)` should do it.

Reply:
👍  https://github.com/scf4/styled-map/pull/19/commits/600924203e910f0419a55683fe9de4dd451a2477